src/server.cpp

```diff
@@ -107,6 +141,16 @@ void connection_handler(int connection_fd, std::size_t source_id){
 
             auto& sensor = sources[source_id].sensors[sensor_id];
 
+	    sensor.data = data;
+
+	    try {
+		CppSQLite3Buffer bufSQL;
+		bufSQL.format("insert into sensor_data (data, fk_sensor) values (%s, %zu);", data.c_str(), sensor_id+1);
```

Comment:
The sensor_id is likely wrong here for fk_sensor. The sensor_id is local to each source in the C++ code, while in the database, it is globally unique for all sensors. (We can discuss that tomorrow in details).